### PR TITLE
[Azure.Monitor.Query] Update APIView

### PR DESCRIPTION
**Pull request by nisha-bhatia**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.Query/src/MetricsClient.cs [hunk 1]
```diff
@@ -69,9 +69,9 @@ protected MetricsClient()
         /// <param name="options">The <see cref="MetricsQueryResourcesOptions"/> to configure the query.</param>
         /// <param name="cancellationToken">The <see cref="CancellationToken"/> to use.</param>
         /// <returns>A time series metrics result for the requested metric names.</returns>
-        public virtual Response<MetricsQueryResourcesResult> QueryResources(IEnumerable<ResourceIdentifier> resourceIds, List<string> metricNames, string metricNamespace, MetricsQueryResourcesOptions options = null, CancellationToken cancellationToken = default)
+        public virtual Response<MetricsQueryResourcesResult> QueryResources(IEnumerable<ResourceIdentifier> resourceIds, IEnumerable<string> metricNames, string metricNamespace, MetricsQueryResourcesOptions options = null, CancellationToken cancellationToken = default)
         {
-            if (resourceIds.Count() == 0 || metricNames.Count == 0)
+            if (resourceIds.Count() == 0 || metricNames.Count() == 0)
```
Comment (tg-msft) [inline sdk/monitor/Azure.Monitor.Query/src/MetricsClient.cs:74]: Nit - should add an `Argument.IsNotNull` for these before calling a method on them

**Review by tg-msft — APPROVED** [review]
Looks good!

**Review by pallavit — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.Query/src/MetricsClient.cs (hunk 1 above)
Comment (nisha-bhatia) [reply sdk/monitor/Azure.Monitor.Query/src/MetricsClient.cs:74]: the lists cannot be null because they are instantiated but I do check if the count is 0

Comment (tg-msft) [reply sdk/monitor/Azure.Monitor.Query/src/MetricsClient.cs:74]: A customer could always do something like:
```csharp
IEnumerable<ResourceIdentifier> ids = null;
if (ReturnsTrueMostOfTheTime())
{
    ids = GetIds();
}
client.QueryResources(ids, ["foo"], "fooNs");
```

It's not critical to fix this release that we want out ASAP, but something we'd ideally fix in the future.  (And it's not a breaking change to go from a NullRefException to an ArgumentNullException.)**

**Pull request merged**